src/Spain/Mews.Fiscalizations.Spain/Model/Response/ResponseResult.cs

```diff
@@ -1,38 +1,41 @@
-﻿using Mews.Fiscalizations.Core.Model;
+﻿using FuncSharp;
+using Mews.Fiscalizations.Core.Model;
 
 namespace Mews.Fiscalizations.Spain.Model.Response
 {
     public static class ResponseResult
     {
         public static ResponseResult<T> Success<T>(T result) where T : class
         {
-            return new ResponseResult<T>(successResult: result);
+            return new ResponseResult<T>(result);
         }
 
         public static ResponseResult<T> Error<T>(ErrorResult error) where T : class
         {
-            return new ResponseResult<T>(errorResult: error);
+            return new ResponseResult<T>(error);
         }
 
         public static ResponseResult<T> Error<T>(string code, string message) where T : class
         {
-            return new ResponseResult<T>(errorResult: new ErrorResult(code, message));
+            return new ResponseResult<T>(new ErrorResult(code, message));
         }
     }
 
-    public class ResponseResult<TResult>
+    public class ResponseResult<TResult> : Coproduct2<TResult, ErrorResult>
         where TResult : class
     {
-        internal ResponseResult(TResult successResult = null, ErrorResult errorResult = null)
+        internal ResponseResult(TResult successResult = null) : base(successResult)
         {
-            SuccessResult = successResult;
-            ErrorResult = errorResult;
         }
 
-        public TResult SuccessResult { get; }
+        internal ResponseResult(ErrorResult errorResult = null) : base(errorResult)
+        {
+        }
+
+        public IOption<TResult> SuccessResult => First;
 
-        public ErrorResult ErrorResult { get; }
+        public IOption<ErrorResult> ErrorResult => Second;
 
-        public bool IsSuccess => SuccessResult.IsNotNull();
+        public bool IsSuccess => IsFirst;
     }
```

Comment:
Not sure i get this, why do we have 2 classes with the same name and one of them is static and the other is a coproduct?

This can be done in a single class

Reply:
In this it is allow type inferring to kick in. At least for Success method. Assuming we have variable "result" of type "ResponseResult<TResult>"

Current, ResponseResult::Success<T> way:

> var result = ResponseResult.Success(result);

Static method on generic class, ResponseResult<T>::Success way:

> var result = ResponseResult<TResult>.Success(result);

I would say suggested implementation is more concise. However I believe there are samples when we do it both ways in codebase. Would you prefer second way?